adapters/v1/syft.go

```
@@ -188,7 +188,14 @@ func (s *SyftAdapter) CreateSBOM(ctx context.Context, name, imageID, imageTag st
 	// download image
 	logger.L().Debug("downloading image", helpers.String("imageID", imageID))
 
-	ctxWithSize := context.WithValue(context.Background(), image.MaxImageSize, s.maxImageSize)
+	ctxWithTimeout := ctx
+	if s.scanTimeout > 0 {
+		var cancel context.CancelFunc
+		ctxWithTimeout, cancel = context.WithTimeout(ctx, s.scanTimeout)
+		defer cancel()
+	}
+
+	ctxWithSize := context.WithValue(ctxWithTimeout, image.MaxImageSize, s.maxImageSize)
```

golangci-lint (2.12.2)

[error] SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)